plugin/storage/kafka/options_test.go

```diff
@@ -45,4 +53,144 @@ func TestFlagDefaults(t *testing.T) {
+
+	compressionLevel, err = getCompressionLevel("snappy", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["snappy"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("lz4", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["lz4"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("zstd", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["zstd"].defaultCompressionLevel, compressionLevel)
+}
+
+func TestGetCompressionLevel(t *testing.T) {
+	compressionLevel, err := getCompressionLevel("none", 0)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["none"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("gzip", 4)
+	require.NoError(t, err)
+	assert.Equal(t, 4, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("snappy", 0)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["snappy"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("lz4", 10)
+	require.NoError(t, err)
+	assert.Equal(t, 10, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("zstd", 20)
+	require.NoError(t, err)
+	assert.Equal(t, 20, compressionLevel)
+}
+
+func TestFailedCompressionLevelScenario(t *testing.T) {
+	_, err := getCompressionLevel("gzip", 14)
+	require.Error(t, err)
+
+	_, err = getCompressionLevel("lz4", 18)
+	require.Error(t, err)
+
+	_, err = getCompressionLevel("zstd", 25)
+	require.Error(t, err)
+}
+
+func TestGetCompressionModes(t *testing.T) {
+	compressionModes, err := getCompressionModes("gzip")
+	require.NoError(t, err)
+	assert.Equal(t, sarama.CompressionGZIP, compressionModes)
+
+	compressionModes, err = getCompressionModes("snappy")
+	require.NoError(t, err)
+	assert.Equal(t, sarama.CompressionSnappy, compressionModes)
+
+	compressionModes, err = getCompressionModes("none")
+	require.NoError(t, err)
+	assert.Equal(t, sarama.CompressionNone, compressionModes)
+}
+
+func TestGetCompressionModesFailures(t *testing.T) {
+	_, err := getCompressionModes("test")
+	require.Error(t, err)
+
+	_, err = getCompressionModes("temp")
+	require.Error(t, err)
+}
+
+func TestGetRequiredAcks(t *testing.T) {
+	requiredAcks, err := getRequiredAcks("noack")
+	require.NoError(t, err)
+	assert.Equal(t, sarama.NoResponse, requiredAcks)
+
+	requiredAcks, err = getRequiredAcks("local")
+	require.NoError(t, err)
+	assert.Equal(t, sarama.WaitForLocal, requiredAcks)
+
+	requiredAcks, err = getRequiredAcks("all")
+	require.NoError(t, err)
+	assert.Equal(t, sarama.WaitForAll, requiredAcks)
+}
+
+func TestGetRequiredAcksFailures(t *testing.T) {
+	_, err := getRequiredAcks("test")
+	require.Error(t, err)
+
+	_, err = getRequiredAcks("temp")
+	require.Error(t, err)
+}
+
+func TestCompressionModes(t *testing.T) {
```

Comment:
I don't think this test is useful, it just tests that map[] works as expected

Reply:
fixed